apps/smart-contracts/token/test/MiniSales.test.ts

```diff
@@ -435,4 +455,221 @@ describe('=> MiniSales', () => {
       await expect(tx).to.emit(miniSales, 'PurchaseHookChange').withArgs(JUNK_ADDRESS)
     })
   })
+
+  describe('# testEndToEnd', () => {
+    let disallowedSourceUser: SignerWithAddress
+    let disallowedDestinationUser: SignerWithAddress
+    let blockedRecipient: SignerWithAddress
+    let ppo: PPO
+    let allowlistPurchaseHook: AllowlistPurchaseHook
+    let sourceAllowlist: AccountList
+    let destinationAllowlist: AccountList
+    let blocklist: AccountList
+    let purchaseAllowlist: AccountList
+    let restrictedTransferHook: RestrictedTransferHook
+    let allowedSourceUser: SignerWithAddress
+    let allowedDestinationUser: SignerWithAddress
+    let blockedUser: SignerWithAddress
+    let unblockedUser1: SignerWithAddress
+
+    const getSignersAndMockTokenForE2ETests = async (): Promise<void> => {
+      await setupSignersAndMockTokens()
+      ;[
+        allowedSourceUser,
+        disallowedSourceUser,
+        allowedDestinationUser,
+        disallowedDestinationUser,
+        blockedUser,
+        unblockedUser1,
+      ] = await ethers.getSigners()
+    }
+
+    const setupUnpermissionedEndToEndStack = async (): Promise<void> => {
```

Comment:
```suggestion
    const setupEndToEndStack = async (): Promise<void> => {
```
No need for this, maybe just add a comment saying sale permissions are setup in a separate function `setupPermissionSale` rather than making this function name unnecessarily long.